sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1837,6 +1841,35 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Get history of application versions for an application
+   * where the current application is excluded.
+   * @param applicationId application id.
+   * @returns assessment history list.
+   */
+  async applicationVersionHistory(
+    applicationId: number,
+  ): Promise<Application[]> {
+    const application = await this.repo.findOne({
+      select: { applicationNumber: true },
+      where: { id: applicationId },
+    });
+    if (!application) {
+      return null;
+    }
+    return this.repo.find({
+      select: { id: true, submittedDate: true },
+      where: {
+        id: Not(applicationId),
+        applicationNumber: application.applicationNumber,
+        applicationStatus: Not(ApplicationStatus.Draft),
```

Comment:
The `applicationStatus: Not(ApplicationStatus.Draft)` does not have any effect as draft applications will not have application number and will be eliminated by previous condition.